### DISCUSSION TITLE
Own FileFormat is registered in the File widget

### DISCUSSION THREAD

**Discussion by mw25** [biolab/orange3#5537] [Q&A] [answered]
Hello,

I wrote an importer widget for an own binary file format. I also introduced a new file format by deriving the FileFormat class.
The new widget works fine, but the browse dialog of the original file widget now also shows my own format, which I don't want. The binary file format should only be able to be opened via the new widget.

Is it intentional to be able to open all registered file formats with the file widget?
Is there any mechanism for a custom FileFormat not to be used by the File widget?

Thanks in advance.

**Comment by markotoplak** [comment, marked as answer]
This was done intentionally; FileFormat keeps a registry. See the implementation of `class _FileFormatMeta`.

As I see there, if you do not implement `read()` (just name it differently), it should not appear in the list readers in OWFIle.

**Reply by mw25** [reply]
Thanks for the quick reply.

I have observed how the _FileFormatMeta class registers the readers.
I have also tried renaming EXTENSIONS. Which I think does the same thing as renaming read.
In fact, it doesn't show up in the file widget then.

However, I like the way 
`reader = FileFormat.get_reader(filepath)`
can be called.
This is then no longer possible. 

But if it is the only option, I can live with it. I can also get the correct FileFormat (there are actually several) in another way.

**Reply by markotoplak** [reply]
I do not see an easy option. Looking backward, I think the solution Orange uses for automatic file reader registration is not ideal. I would prefer something explicit instead of implicit.

**Reply by mw25** [reply]
Is it possible in principle to have a second registry with my own formats?

The following post gave me the idea to derive and redeclare `Registry`/`_FileFormatMeta`/`FileFormatBase` to achieve this. 
https://stackoverflow.com/a/64146346
But unlike there, I just want a new registry and not change the behavior of the metaclass.
